### PR TITLE
scripts/docker: Make containerized build work with podman

### PR DESCRIPTION
**Pull request by osteffenrh**
- Detect podman and set special options for user mapping and SELinux
- Make build.sh work with spaces in file paths
- Make script return codes consistent

### CODE REVIEW

**Comment by osteffenrh** [comment]
Rebased, improved detection of podman, added some other fixes for build.sh

**Comment by osteffenrh** [comment]
@stefano-garzarella, here is v2 :-)

**Comment by stefano-garzarella** [comment]
@osteffenrh great!

Tested on:
- a F38 with `podman` and `podman-docker` installed: worked
- a F38 with only `docker-ce` installed: worked

Just some very small comments:
- in the last commit title patch, maybe there is a typo (s/cores/codes ?)
- should we now rename `script/docker` folder in `script/container` ?

**Review by stefano-garzarella — APPROVED** [review]

**Review by cclaudio — COMMENTED** [review]
I suggested only one small change, without it the `build.sh` fails with docker.

Other than that, the PR looks good to me. Thanks!

File: scripts/container/build.sh
[the remark this answers is not in the document]
Comment (osteffenrh) [reply scripts/container/build.sh:1]: of course. I somehow did not rebuild the image and missed this.

**Review by stefano-garzarella — APPROVED** [review]

**Review by cclaudio — APPROVED** [review]
Great, thanks.

**Pull request merged**